src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -1226,16 +1226,17 @@ private void EnsureParameterInitialized(TaskPropertyInfo parameter, Lookup looku
 
             parameter.Initialized = true;
 
-            string taskAndParameterName = _taskName + "_" + parameter.Name;
-            string key = "DisableLogTaskParameter_" + taskAndParameterName;
+            // PERF: Be careful to avoid unnecessary string allocations. Appending '_taskName + "_" + parameter.Name' happens in both paths,
+            // but we don't want to allocate the string if we don't need to.
+            string key = "DisableLogTaskParameter_" + _taskName + "_" + parameter.Name;
```

Comment:
If this is causing noticeable allocations, it looks like we could avoid them entirely by building a lookup table at the beginning of execution (after evaluation) and looking up here without constructing a string.

Reply:
How are you building Roslyn? Is more logging turned on than you expect? Because I am a bit surprised "log task inputs" is doing anything.